### PR TITLE
Check for NonStructuralSchema with kind

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/9386

It's possible to create a CRD which is valid (even with generated CRDs) but is not accepted by the Kubernetes API due to `NonStructuralSchema`. The recommended way to test this is to actually install the CRD on a running cluster so I have implemented this as a sort of integration test using kind.

## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [x] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,9 +3,49 @@ version: 2.1
...
+          command: |
+            curl -sLo kind https://github.com/kubernetes-sigs/kind/releases/download/$KIND_VERSION/kind-linux-amd64
+      - run:
+          name: Download kubectl
+          command: |
+            curl -sLO https://storage.googleapis.com/kubernetes-release/release/$KUBERNETES_VERSION/bin/linux/amd64/kubectl && chmod +x kind kubectl
+      - run:
+          name: Create cluster
+          command: |
+            ./kind create cluster --image quay.io/giantswarm/kind-node:$KUBERNETES_VERSION --name apiextensions 2> /dev/null
```
Comment (MarcelMue) [inline .circleci/config.yml:25]: Is there some idea to align here between projects? I am not against this - just wondering as I fear divergence.

cc @giantswarm/wg-integration-testing

Comment (tfussell) [reply .circleci/config.yml:25]: Same here. I was having trouble determining the current state-of-the-art when it comes to testing with kind so I decided to take the straightforward approach. If we have some tools that can do this, that would be cool.

Comment (xh3b4sd) [reply .circleci/config.yml:25]: I think this goes into the direction of getting the future vision right which SIG RelEng may want to think about.

**Review by xh3b4sd — APPROVED** [review]
This should help a great deal already. I guess you did some testing and broke some CRDs purposefully to see the test job fail. So when this works then let's roll.

**Comment by tfussell** [comment]
> This should help a great deal already. I guess you did some testing and broke some CRDs purposefully to see the test job fail. So when this works then let's roll.

Yes, it's actually quite difficult to create broken CRDs now with all of the code generation, unit tests, and CI, but here's is an example of a failure I induced using kustomize-based post-processing of CRDs to add validation to `.metadata.creationTimestamp` which isn't allowed https://circleci.com/gh/giantswarm/apiextensions/2309

**Pull request merged**